app/models/file_version_checker.rb

```diff
@@ -35,8 +35,11 @@ def process_content
 
       begin
         reader = PDF::Reader.new(file_path)
+        if contains_arxiv_watermark?(reader)
+          @score = -9999
+          return ''
+        end
```

Comment:
I think it would probably be good to pull `reader = PDF::Reader.new(file_path)` out of this and memoize it in its own function: 
```
def reader
   @reader ||= PDF::Reader.new(file_path)
end
```

Then, move `if contains_arxiv_watermark?(reader)` up to the `version` method, and have it return accepted version.  Forcing the score down to -9999 feels like a work around that I don't think is necessary, but I don't know exactly how the PDF::Reader object operates.  I could be wrong if it operates like an enumerator.